src/script/squirrel_helper.hpp

```diff
@@ -52,6 +48,16 @@ namespace SQConvert {
 		}
 	};
 
+	template <ConvertibleThroughBase T> struct Return<T> {
+		static inline int Set(HSQUIRRELVM vm, T res)
+		{
+			/* Squirrel wants all types to be signed. */
+			using signed_type = typename std::make_signed_t<typename T::BaseType>;
```

Comment:
I'm not sur this is correct if base type is `uint8_t` or `uint16_t` (when looking at corresponding `Return<>` above).
All current `StrongType`s except `GRFStringID` are 32bit so not really an issue, but it could be.

Reply:
This would make `INVALID_STATION` and friends negative, if it were to be migrated when they wouldn't be at the moment.